packages/vue/src/vue-component-lib/overlays.ts

```diff
@@ -162,10 +162,22 @@ export const defineOverlayContainer = <Props extends object>(name: string, defin
           }
         }
 
+        /**
+         * `ion-modal` needs a wrapper element so content
+         * takes up the full size of the parent modal.
+         */
+        const renderChildren = () => {
+          if (name === 'ion-modal') {
```

Comment:
~ I'm not personally a huge fan of coupling component specific checks in the shared utility for all overlays. 

Would you be open to refactor this to be an option of `defineOverlayContainer` and update the manual wrappers to set the option: https://github.com/ionic-team/ionic-framework/blob/a1e3e4ef04226fdd661689ce6f0e8064d8ed28d2/packages/vue/src/components/Overlays.ts#L30?

Reply:
I'm open to changing the approach, though I don't follow the proposed alternative. Can you clarify?

Reply:
For sure!

Updating the implementation to:

```ts
export const defineOverlayContainer = <Props extends object>(name: string, defineCustomElement: () => void, componentProps: string[] = [], controller?: any, delegateHost: boolean) => {
  ...
  const renderChildren = () => {
    if(delegateHost) {
      return h('div', { className: 'ion-delegate-host ion-page' }, slots)
    }
    return slots;
  }
}
```

Making the usage for the overlay wrappers:
```ts
export const IonModal = /*@__PURE__*/ defineOverlayContainer<JSX.IonModal>('ion-modal', defineIonModalCustomElement, ['props'], true);
    
```

This may be an advantageous time to refactor `defineOverlayContainer` to take an object of configurations, vs. many different arguments for configuration. 

e.g.:
```ts
defineOverlayContainer<JSX.IonModal>({
  tag: 'ion-modal',
  defineCustomElement: defineIonModalCustomElement,
  props: [],
  hasDelegateHost: true
});
```

but that could also be a separate discussion/work-effort, personally many arguments can be hard to reason about.

The whole goal here of introducing a new argument, is that the responsibility of rendering the `ion-delegate-host ion-page` container is independent of the tag name and can be enabled for any overlay that needs it.

Reply:
Ok I made the changes. Lmk your thoughts.